### ISSUE TITLE
Feature request: simple function to get the IPFS hash for a local file

### ISSUE THREAD

**Issue by davux** [ipfs/js-ipfs#1799]
#### Description:

The mobile app I'm working on needs to store local files. I would like to name those files after their IPFS hash so that I don't store the same file twice. Running an IPFS node is overkill because of resource consumption on a mobile device. All I want is to name files after their IPFS hash (rather than their hex-encoded SHA-something hash). Multihash, and specifically IPFS naming compatibility is still useful for present and future compatibility with other aspects of the project.

In summary, it would be very useful to have a simple static function that returns the IPFS hash of a given file based on its content and/or its local path.

**Comment by alanshaw** [comment]
@davux you can use the `onlyHash` option with `ipfs.add` to get back just the hash.

You can use this with `js-ipfs` or `js-ipfs-http-client` if you're talking to a remote IPFS node.

If that's not possible for whatever reason you need to look into using the [IPFS unixfs importer](https://github.com/ipfs/js-ipfs-unixfs-importer) directly with the `onlyHash` option.

One thing I don't understand - if you don't add the file to IPFS it'll not be accessiible anywhere so what use is the hash?

**Comment by alanshaw** [comment]
So, does the `onlyHash` option answer you question?

**Comment by davux** [comment]
Thanks @alanshaw for the pointers, that does the job.

To answer your question: in some cases the file is to be published on IPFS, in other cases it isn't, and I don't want 2 naming schemes for local copies. At any rate, if running IPFS on mobile was an option I would go for that but it looks like that's still in mature stages for now, at least for react-native.

Thanks again for your help. I'm closing this issue now but you might want to make the documentation clearer for that use case, as I've seen the same question a few times. Google will gladly end up pointing people to this issue, but I guess that's not really how it should work. :)

**Comment by alanshaw** [comment]
Hey @davux - I published this last night: https://www.npmjs.com/package/ipfs-only-hash

It's just the code you need to calculate an IPFS hash without having to install or start a JS IPFS node. Feedback welcome!

**Comment by fawazahmed0** [comment]
Sharing this, might help someone:
```js
const IPFS =  require('ipfs-core')
const ipfs = await IPFS.create({start: false})
const { cid } = await ipfs.add('Hello world',{onlyHash:true,cidVersion:1,hashAlg:'blake2b-256'})
console.info(cid.toString())
```
